feat: platform onboarding flow and dashboard

Ryukemeister opened the pull request:
## What does this PR do?

<!-- Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change. -->

This PR adds a new flow for creating a platform client, previously we were using the flow that is used to create an organization. This PR fixes that and also adds a new dashboard for a platform user under which they can manage their oauth clients (create, update or delete) and see a list of their managed users. There are also other tabs for seeing platform docs, and other examples. This PR also includes billing in the app itself using stripe.

### Steps to create a new platform client: 
- Go to `/settings/platform/new` and create a new platform user
- Once that is done you are redirected to the platform dashboard, initially you'll see all the pricing plans for platform.
- Once you subscribe to any of the plans through stripe you get redirected back to the actual dashboard. This dashboard has a list of your oauth client, managed users and some more resources you might need in order to get started with platform.


@PeerRich: also try:

https://localhost:3000/signup?redirect=https://app.cal.com/settings/platform/new

Loom: https://www.loom.com/share/eb02cfc3aa654e07b29c90a95c93fac4

Ryukemeister committed:
add endpoint to fetch managed user from client id

Ryukemeister committed:
update typings

Ryukemeister committed:
minor tweaks

Ryukemeister committed:
custom hook to fetch managed users from client id

Ryukemeister committed:
add translations for platform onboarding

Ryukemeister committed:
add isPlatformOrg boolean to figure out which is platform and which is not

Ryukemeister committed:
set isPlatform hook based on data obtained from org

Ryukemeister committed:
add limitWidth prop to control component width

Ryukemeister committed:
add props to shell to make sidebar display different tabs based on if the shell isPlattform or not

Ryukemeister committed:
platform related pages

Ryukemeister committed:
fix merge conflicts

Ryukemeister committed:
fix merge conflicts

Ryukemeister committed:
platform oauth client form and card

Ryukemeister committed:
Merge branch 'main' into platform-onboarding

Ryukemeister committed:
Merge branch 'main' into platform-onboarding

PeerRich commented:
can you do a self review and loom video?

will billing be merged into this or separate @exception

Ryukemeister commented:
> can you do a self review and loom video?
> 
> will billing be merged into this or separate @exception

Will do a self review today and also leave a link to the loom video.

Ryukemeister committed:
remove everything related to platform from organization

Ryukemeister committed:
update oauth client card and form

Ryukemeister committed:
fixup

Ryukemeister committed:
fix imports and remove logs

Ryukemeister committed:
fixup

exception commented:
> can you do a self review and loom video?
> 
> will billing be merged into this or separate @exception

Seperate.

exception commented on apps/api/v2/src/modules/oauth-clients/controllers/oauth-clients/oauth-clients.controller.ts:
```diff
@@ -98,6 +107,21 @@ export class OAuthClientsController {
 
+  @Get("/managed-users/:clientId")
+  @HttpCode(HttpStatus.OK)
+  @Roles([MembershipRole.ADMIN, MembershipRole.OWNER, MembershipRole.MEMBER])
```
If MEMBER is required, do we need the others? cc @supalarry

ThyMinimalDev replied:
looking at the code of apps/api/v2/src/modules/auth/guards/organization-roles/organization-roles.guard.ts it is required yes

supalarry replied:
As Morgan noted above - we indeed need to explicitly list all roles.

Ryukemeister committed:
update redirect url

Ryukemeister committed:
split oauth client form into separate update and create forms

Ryukemeister committed:
separate forms for create and edit oauth clients

Ryukemeister committed:
fixup

Ryukemeister commented on packages/features/ee/organizations/components/CreateANewOrganizationForm.tsx:
```diff
@@ -24,15 +24,21 @@ function extractDomainFromEmail(email: string) {
 }
 
-export const CreateANewOrganizationForm = () => {
+export const CreateANewOrganizationForm = ({ isPlatformOrg = false }: { isPlatformOrg?: boolean }) => {
```
`CreateANewOrganizationForm` takes in another optional prop `isPlatformOrg`, the purpose of this prop is to detect if this form is being used in the org flow or platform flow since org flow has 4 steps and platform flow just requires 1 step. Also a lot of stuff from the form like org url and seats and other stuff is not needed in platform so the `isPlatformOrg` conditionally renders components.

Ryukemeister commented on packages/lib/server/repository/organization.ts:
```diff
@@ -53,6 +53,7 @@ export class OrganizationRepository {
           orgPricePerSeat: orgData.pricePerSeat,
           isPlatform: orgData.isPlatform,
         },
+        isPlatform: orgData.isPlatform,
```
This is very important, since what is happening currently is the `isPlatform` boolean that comes from the orgData in our frontend stores the boolean in the metadata instead of setting the `isPlatform` table boolean to the given value . As a reason even after create an org for platform customers we have to manually go into db and set the `isPlatform` hook to true. The above line of code fixes that, so we don't have to manually go into the db from now on to set that book to true.

Ryukemeister commented on packages/ui/components/empty-screen/EmptyScreen.tsx:
```diff
@@ -21,6 +21,7 @@ export function EmptyScreen({
   className,
   iconClassName,
   iconWrapperClassName,
+  limitWidth = true,
```
This is to not limit the width of the empty screen which was looking better in the platform dashboard.

Ryukemeister commented on packages/features/shell/Shell.tsx:
```diff
@@ -648,27 +649,72 @@ const navigation: NavigationItemType[] = [
-      items.mobileNavigationMoreItems.push(item);
-    }
-    return items;
+const platformNavigation: NavigationItemType[] = [
```
This should be the sidebar layout for platform dashboard.

Ryukemeister commented on packages/features/shell/Shell.tsx:
```diff
@@ -648,27 +649,72 @@ const navigation: NavigationItemType[] = [
+  },
+];
+
+const getDesktopNavigationItems = (isPlatformNavigation = false) => {
```
Previously we were using a standard sidebar layout for all the pages i.e. all sidebars are the same for every page. Since the platform dashboard requires a different layout and set of tabs this function checks if the navigation sidebar is for platform customer or not and returns items accordingly.

Ryukemeister commented on packages/features/shell/Shell.tsx:
```diff
@@ -948,54 +999,54 @@ function SideBar({ bannersHeight, user }: SideBarProps) {
-          ))}
-          {!IS_VISUAL_REGRESSION_TESTING && <Credits />}
-        </div>
+        {!isPlatformUser && (
```
Platform dashboard doesnt need the lower sidebar containing view public page, OOO banner so thats why it conditionally renders now.

Ryukemeister committed:
Merge branch 'main' into platform-onboarding

ThyMinimalDev commented on apps/web/pages/settings/platform/oauth-clients/edit.tsx:
```diff
@@ -0,0 +1,32 @@
+          <div className="border-subtle mx-auto block justify-between rounded-t-lg border px-4 py-6 sm:flex sm:px-6">
+            <div className="flex w-full flex-col">
+              <h1 className="font-cal text-emphasis mb-1 text-xl font-semibold leading-5 tracking-wide">
+                OAuth client {!!clientId ? "updation" : "creation"} form
```
this should always be for updating

ThyMinimalDev commented on apps/web/pages/settings/platform/oauth-clients/edit.tsx:
```diff
@@ -0,0 +1,32 @@
+
+export default function CreateOAuthClient() {
+  const searchParams = useCompatSearchParams();
+  const clientId = searchParams?.get("clientId") || "";
```
I would use a dynamic param and reorganize, instead of getting the clientId from searchparams, use next.js routing 
and have something like this /platform/oauth-clients/:OAUTH_CLIENT_ID/edit

ThyMinimalDev commented on apps/web/pages/settings/platform/index.tsx:
```diff
@@ -0,0 +1,237 @@
+                      className="border-subtle overflow-hidden rounded-b-lg border border-b-0"
+                      span={3}
+                    />
+                    <tr>
+                      <td className="border-subtle border px-4 py-3 md:text-center">Id</td>
+                      <td className="border-subtle border px-4 py-3 md:text-center">Username</td>
+                      <td className="border-subtle border px-4 py-3 md:text-center">Email</td>
+                    </tr>
+                    {managedUserData.map((user) => {
+                      return (
+                        <tr key={user.id} className="">
+                          <td className="border-subtle overflow-hidden border px-4 py-3 md:text-center">
+                            {user.id}
+                          </td>
+                          <td className="border-subtle border px-4 py-3 md:text-center">{user.username}</td>
+                          <td className="border-subtle overflow-hidden border px-4 py-3 md:overflow-auto md:text-center">
+                            {user.email}
+                          </td>
+                        </tr>
+                      );
+                    })}
+                  </table>
+                </>
+              ) : (
+                <EmptyScreen
+                  limitWidth={false}
+                  headline={
+                    initialClientId == undefined
+                      ? "OAuth client is missing. You need to create an OAuth client first in order to create a managed user."
+                      : `OAuth client ${initialClientId} does not have a managed user present.`
+                  }
+                  description={
+                    initialClientId == undefined
+                      ? "Refer to the Platform Docs from the sidebar in order to create an OAuth client."
+                      : "Refer to the Platform Docs from the sidebar in order to create a managed user."
+                  }
+                  className="items-center border"
+                />
+              )}
+            </div>
```
move this into its own component

ThyMinimalDev replied:
try to follow have a smart container (ManagedUsers) rendering dumb children components

ThyMinimalDev commented on packages/features/ee/organizations/components/CreateANewOrganizationForm.tsx:
```diff
@@ -94,7 +106,11 @@ const CreateANewOrganizationFormChild = ({ session }: { session: Ensure<SessionC
-            createOrganizationMutation.mutate(v);
+            createOrganizationMutation.mutate({
+              ...v,
+              isPlatform: isPlatformOrg ? true : false,
```
isPlatform is already a boolean, no need to use a ternary

ThyMinimalDev commented on apps/web/pages/settings/platform/index.tsx:
```diff
@@ -0,0 +1,237 @@
+                      className="border-subtle overflow-hidden rounded-b-lg border border-b-0"
+                      span={3}
+                    />
+                    <tr>
+                      <td className="border-subtle border px-4 py-3 md:text-center">Id</td>
+                      <td className="border-subtle border px-4 py-3 md:text-center">Username</td>
+                      <td className="border-subtle border px-4 py-3 md:text-center">Email</td>
+                    </tr>
+                    {managedUserData.map((user) => {
+                      return (
+                        <tr key={user.id} className="">
+                          <td className="border-subtle overflow-hidden border px-4 py-3 md:text-center">
+                            {user.id}
+                          </td>
+                          <td className="border-subtle border px-4 py-3 md:text-center">{user.username}</td>
+                          <td className="border-subtle overflow-hidden border px-4 py-3 md:overflow-auto md:text-center">
+                            {user.email}
+                          </td>
+                        </tr>
+                      );
+                    })}
+                  </table>
+                </>
+              ) : (
+                <EmptyScreen
+                  limitWidth={false}
+                  headline={
+                    initialClientId == undefined
+                      ? "OAuth client is missing. You need to create an OAuth client first in order to create a managed user."
+                      : `OAuth client ${initialClientId} does not have a managed user present.`
+                  }
+                  description={
+                    initialClientId == undefined
+                      ? "Refer to the Platform Docs from the sidebar in order to create an OAuth client."
+                      : "Refer to the Platform Docs from the sidebar in order to create a managed user."
+                  }
+                  className="items-center border"
+                />
+              )}
+            </div>
```
move into it's own component ManagedUsers, rendering children components: OAuthClientSelector, ManagedUsersList,...

ThyMinimalDev commented on apps/web/pages/settings/platform/index.tsx:
```diff
@@ -0,0 +1,237 @@
+                        <Button color="secondary">{initialClientName}</Button>
+                      </DropdownMenuTrigger>
+                      <DropdownMenuContent>
+                        {data.map((user) => {
```
user ? the data is containing oAuthCLients no ?

Ryukemeister replied:
The data here contains managed users of a particular oauth client, thats why the user

ThyMinimalDev replied:
the data here seems to be an array of OAuthClients, isn't this a dropdown select to choose which oauth clients you want to see the managed users of ?

Ryukemeister replied:
Ah yes right, my bad sorry.

PeerRich commented:
missing loom, self review and PR description how to test

Ryukemeister commented:
> missing loom, self review and PR description how to test

@PeerRich self review is done, description is updated and loom video is on the way

Ryukemeister commented:
Loom: https://www.loom.com/share/264d67dd63f140fbbd6e54a5f7b80659?sid=d02febb7-e773-44b0-b360-f1f16d0506c1

Ryukemeister committed:
fixup

Ryukemeister committed:
dynamic routes for oauth client edit page

ThyMinimalDev committed:
Merge branch 'main' into platform-onboarding

ThyMinimalDev commented on apps/web/components/settings/platform/oauth-clients/oauth-client-form/edit.tsx:
```diff
@@ -337,21 +307,17 @@ export const OAuthClientForm: FC<{ clientId?: string }> = ({ clientId }) => {
-            </Tooltip>
-            <Button type="button" onClick={selectAllPermissions} disabled={disabledForm || Boolean(clientId)}>
+            <h1 className="text-base font-semibold underline">Permissions</h1>
+            <Button type="button" onClick={selectAllPermissions}>
```
shouldn't be able to edit permissions of an already existing oauth client

Ryukemeister committed:
fixup fixup

ThyMinimalDev commented on apps/web/components/settings/platform/oauth-clients/oauth-client-form/create.tsx:
```diff
@@ -0,0 +1,273 @@
+  areEmailsEnabled?: boolean;
+};
+
+export const OAuthClientForm = () => {
```
there is a lot of duplicated code between create.tsx and edit.tsx, I would create an extra component called Fields containing the rendering part and whatever can be mutualized between create and edit

Ryukemeister committed:
fix to not show error when redirect uri is empty

ThyMinimalDev commented on apps/web/pages/settings/platform/oauth-clients/[clientId]/edit/index.tsx:
```diff
@@ -0,0 +1,33 @@
+export default function EditOAuthClient() {
+  const params = useParams<{ clientId: string }>();
+  const clientId = params?.clientId || "";
+
```
get the oAuthClient Data using the hook

ThyMinimalDev replied:
get the update function from the hook, and create handleSubmit function

joeauyeung reviewed (commented):
@Ryukemeister I just reviewed the new org and shell files.

exception committed:
Merge remote-tracking branch 'origin/main' into platform-onboarding

# Conflicts:
#	apps/api/v2/src/modules/oauth-clients/controllers/oauth-clients/oauth-clients.controller.ts

Ryukemeister committed:
refactor create handler for org

Ryukemeister committed:
cleaup comments

CarinaWolli committed:
Merge branch 'main' into platform-onboarding

Ryukemeister committed:
add custom hook to check user billing

Ryukemeister committed:
export managed user type

Ryukemeister committed:
refactor platform index page

Ryukemeister committed:
refactor edit and create pages

Ryukemeister committed:
dashboard component containing oauth client list and managed user

Ryukemeister committed:
common oauth client form used for create and edit form

Ryukemeister committed:
platform pricing helper

Ryukemeister committed:
platform pricing component

Ryukemeister committed:
Merge branch 'main' into platform-onboarding

Ryukemeister committed:
fix typing and data response for billing

Ryukemeister committed:
use custom hook to check team billing info

Ryukemeister committed:
fix type checks

Ryukemeister committed:
upgrade conditional rendering for upgrade to org banner

Ryukemeister committed:
add isLoading prop to check button loading state

Ryukemeister committed:
pass in button handler

Ryukemeister committed:
add custom hook to subscribe to stripe and typings

Ryukemeister committed:
update typings

Ryukemeister committed:
fix incorrect endpoint

Ryukemeister committed:
pass in team id as prop

Ryukemeister committed:
fix type check

Ryukemeister committed:
Merge branch 'main' into platform-onboarding

Ryukemeister committed:
update stripe success and cancel redirect url

Ryukemeister committed:
add and pass redirect url param to custom hook

Ryukemeister committed:
Merge branch 'main' into platform-onboarding

Ryukemeister committed:
custom hooks for platform

Ryukemeister committed:
cleanup

Ryukemeister committed:
update imports

Ryukemeister committed:
fix merge conflicts

Ryukemeister committed:
fixup

Ryukemeister committed:
fixup fixup

Ryukemeister committed:
fixup

Ryukemeister committed:
merge conflicts fixup

Ryukemeister committed:
merge conlficts battle :(

Ryukemeister committed:
Merge branch 'main' into platform-onboarding

Ryukemeister committed:
minor fixes

Ryukemeister committed:
skip admin checks for a platform client

keithwillcode committed:
Merge branch 'main' into platform-onboarding

Ryukemeister committed:
Merge branch 'main' into platform-onboarding

PeerRich committed:
Merge branch 'main' into platform-onboarding

Ryukemeister committed:
Merge branch 'main' into platform-onboarding

ThyMinimalDev commented on apps/web/components/settings/platform/dashboard/managed-user-list/index.tsx:
```diff
@@ -0,0 +1,84 @@
+  handleChange: (clientId: string, clientName: string) => void;
+};
+
+export const MangedUserList = ({
```
typo, Manged instead of Managed

ThyMinimalDev commented on apps/web/pages/settings/platform/oauth-clients/[clientId]/edit/index.tsx:
```diff
@@ -0,0 +1,116 @@
+  hasScheduleWritePermission,
+} from "../../../../../../../../packages/platform/utils/permissions";
+
+export default function EditOAuthClient() {
```
CreateOAuthClient here no ?

ThyMinimalDev commented on packages/features/ee/organizations/components/CreateANewOrganizationForm.tsx:
@Ryukemeister what's the status on this ?

Ryukemeister replied:
> @Ryukemeister what's the status on this ?

This has been fixed

Ryukemeister committed:
fix typo

Ryukemeister committed:
append slug with _platform for a platform user

supalarry commented on apps/api/v2/src/modules/oauth-clients/controllers/oauth-clients/oauth-clients.controller.ts:
```diff
@@ -109,6 +116,21 @@ export class OAuthClientsController {
     return { status: SUCCESS_STATUS, data: client };
   }
 
+  @Get("/managed-users/:clientId")
```
I think it makes sense to have `@Get("/:clientId/managed-users")` because IMO users can be seen as sub-resources of an OAuth client aka OAuth client is higher up in the hierarchy.

supalarry commented on apps/api/v2/src/modules/oauth-clients/controllers/oauth-clients/oauth-clients.controller.ts:
```diff
@@ -109,6 +116,21 @@ export class OAuthClientsController {
+  async getOAuthClientManagedUsersById(
+    @Param("clientId") clientId: string
+  ): Promise<GetOAuthClientManagedUsersResponseDto> {
+    const existingManagedUsers = await this.userRepository.findManagedUsersByOAuthClientId(clientId, 0, 50); // second argument is for offset while third is for limit
```
We should have limit and offset as query parameters using "@Query" decorator. Example can be seen in "getBookings" of "https://github.com/calcom/cal.com/blob/1cfd2f705306ccf8c6234e600ae6b74547f80872/apps/api/v2/src/ee/bookings/controllers/bookings.controller.ts" and then remove comment "// second argument is for offset while third is for limit"

supalarry replied:
Hmm, why do we need to add the "getOAuthClientManagedUsersById" endpoint? In "oauth-client-users.controller.ts" we have identical one:

<img width="1009" alt="Screenshot 2024-05-07 at 15 13 03" src="https://github.com/calcom/cal.com/assets/42170848/718a15aa-9de3-47f4-bd68-46a72def956f">

supalarry commented on apps/api/v2/src/modules/oauth-clients/controllers/oauth-clients/oauth-clients.controller.ts:
```diff
@@ -109,6 +116,21 @@ export class OAuthClientsController {
+  ): Promise<GetOAuthClientManagedUsersResponseDto> {
+    const existingManagedUsers = await this.userRepository.findManagedUsersByOAuthClientId(clientId, 0, 50); // second argument is for offset while third is for limit
+
+    if (!existingManagedUsers) {
```
We don't need `if (!existingManagedUsers)` check because it can't be nullish. Here is its return type. If it's empty array its fine we just return empty array as data.

<img width="586" alt="Screenshot 2024-05-07 at 15 09 40" src="https://github.com/calcom/cal.com/assets/42170848/6298795d-bbd5-46f2-9360-6cc2e00f16f8">

Ryukemeister committed:
Merge branch 'main' into platform-onboarding

supalarry commented on apps/web/components/settings/platform/dashboard/managed-user-list/index.tsx:
```diff
@@ -0,0 +1,84 @@
+            See all the managed users created by your OAuth client.
+          </p>
+        </div>
+        {Array.isArray(oauthClients) && oauthClients.length > 0 && (
```
You can do the check `Array.isArray(oauthClients) && oauthClients.length > 0` inside `OAuthClientsDropdown` component - if the check is not met then `OAuthClientsDropdown` can return `null`.

supalarry commented on apps/web/components/settings/platform/dashboard/managed-user-list/index.tsx:
```diff
@@ -0,0 +1,84 @@
+          />
+        )}
+      </div>
+      {Array.isArray(managedUsers) && !isManagedUserLoading && managedUsers.length ? (
```
Code from line 44 to 82 could be a separate component e.g. `<Users>` and then code above from line 26 to 43 `<Header>` or something - this is just a reminder that its easier to read code when its broken down into smaller parts / components.

supalarry commented on apps/web/pages/settings/platform/oauth-clients/create.tsx:
```diff
@@ -0,0 +1,75 @@
+
+import PageWrapper from "@components/PageWrapper";
+import type { FormValues } from "@components/settings/platform/oauth-clients/oauth-client-form";
+import { OAuthClientForm as CreateOAuthClientForm } from "@components/settings/platform/oauth-clients/oauth-client-form";
```
I think it's fine to call it `OAuthClientForm` because we use it also to edit oauth clients not only create.

supalarry commented on packages/features/ee/organizations/components/CreateANewOrganizationForm.tsx:
```diff
@@ -65,10 +73,14 @@ const CreateANewOrganizationFormChild = ({ session }: { session: Ensure<SessionC
       }
-      router.push(`/settings/organizations/${data.organizationId}/about`);
+      router.push(
+        !isPlatformOrg ? `/settings/organizations/${data.organizationId}/about` : "/settings/platform"
```
`!isPlatformOrg` appears in 7 places in this file which to me indicates that platform creation form should be in a separate file - if its just 1 step it greatly differs from normal org IMO. What do you think?

Ryukemeister replied:
Thinking about it now, yes its actually a good idea!

supalarry commented on packages/features/ee/organizations/components/CreateANewOrganizationForm.tsx:
```diff
@@ -93,7 +105,13 @@ const CreateANewOrganizationFormChild = ({ session }: { session: Ensure<SessionC
+            createOrganizationMutation.mutate({
+              ...v,
+              isPlatform: isPlatformOrg,
+              // we dont have checks to see if an organization url is taken or not if it is a platform user
```
I think having to add comment `// we dont have checks to see if an organization url is taken or not if it is a platform user` confirms what I wrote above that maybe its best to have a separate form file for platform.

supalarry commented:
Amazing work Rajiv! 🔥

emrysal reviewed (dismissed):
There's a few typo's and copy mistakes in the platformPlans - other than that looks reasonable 🙏

Ryukemeister commented:
> There's a few typo's and copy mistakes in the platformPlans - other than that looks reasonable 🙏

My bad will sort these out, thanks a lot @emrysal !!!

Ryukemeister commented:
> Amazing work Rajiv! 🔥

Thanks a ton for the review Lauris, will implement the feedback from your review so we can merge this finally!

PeerRich committed:
Merge branch 'main' into platform-onboarding

Ryukemeister committed:
PR feedback

Ryukemeister committed:
dashboard refactor

Ryukemeister committed:
bring back org form to its orginal state

Ryukemeister committed:
add platform folder to ee

Ryukemeister committed:
update typings

Ryukemeister committed:
use new create platform form

Ryukemeister committed:
fixup

Ryukemeister committed:
fix typo and update plans

Ryukemeister committed:
Merge branch 'main' into platform-onboarding

supalarry commented on apps/web/components/settings/platform/dashboard/managed-user-table/index.tsx:
```diff
@@ -0,0 +1,58 @@
+}: ManagedUserTableProps) => {
+  return (
+    <div>
+      {Array.isArray(managedUsers) && !isManagedUserLoading && managedUsers.length ? (
```
Thats just a small tip though - you have done great job! : )

Ryukemeister committed:
simplifying rendering

ThyMinimalDev committed:
Merge branch 'main' into platform-onboarding

Ryukemeister committed:
remove managed users endpoint since it already exists

Ryukemeister committed:
url for endpoint

Ryukemeister committed:
Merge branch 'main' into platform-onboarding

supalarry reviewed (changes requested):
## [MEDIUM] plan column of PlatformBilling

I saw in your loom and also tried myself - after finishing stripe payment the `plan` is `none` - shouldn't it be one of the `PlatformPlan` enum values?

<img width="748" alt="Screenshot 2024-05-08 at 19 13 01" src="https://github.com/calcom/cal.com/assets/42170848/6702ba17-1dfe-4b71-9830-1e408e219f4c">


## [LOW] Sidebar
Personally, I would remove "Cal.com atoms" entry from sidebar and just rename "Platform docs" to "Documentation". Also, instead of having "Examples app" in the sidebar I would just put it in docs - I don't know how much value do people have of having a link to examples app because docs should mention it.

<img width="1706" alt="Screenshot 2024-05-08 at 19 17 28" src="https://github.com/calcom/cal.com/assets/42170848/59ba867c-fbcc-4d2b-858d-a0601cbb4361">

## [HIGH] OAuth client users not appearing?

I created OAuth client and then created managed user for it via API- managed user email has the same OAuth client ID as in dashboard below. However, after creating the user I refresh "settings/platform" and the managed user is not appearing.

Create managed user:
<img width="1285" alt="Screenshot 2024-05-08 at 19 24 48" src="https://github.com/calcom/cal.com/assets/42170848/9cde4ec8-d90c-41d0-84e4-fbf353b7ae85">

It's not appearing:
<img width="1709" alt="Screenshot 2024-05-08 at 19 26 35" src="https://github.com/calcom/cal.com/assets/42170848/6a4e158b-e9d1-43ac-85da-97fe6a860b97">

Does OAuth client users show up for you under the "Managed users" ?

ThyMinimalDev commented on apps/web/components/settings/platform/oauth-clients/oauth-client-form/create.tsx:
```diff
@@ -0,0 +1,273 @@
+    });
+  };
+
+  const isPending = isSaving;
```
not needed anymore, you can use isPending from useCreateOAuthClient directly

ThyMinimalDev commented on apps/web/components/settings/platform/oauth-clients/oauth-client-form/edit.tsx:
```diff
@@ -139,31 +127,18 @@ export const OAuthClientForm: FC<{ clientId?: string }> = ({ clientId }) => {
   };
-  const isPending = isSaving || isUpdating;
+
+  const isPending = isUpdating;
```
not needed anymore, just use isPending from useUpdateOAuthClient directly

Ryukemeister committed:
rename tabs

Ryukemeister committed:
pr feedback

Ryukemeister committed:
managed users endpoint

Ryukemeister committed:
update endpoint

Ryukemeister committed:
remove form

Ryukemeister committed:
Merge branch 'main' into platform-onboarding

Ryukemeister commented:
> ## [MEDIUM] plan column of PlatformBilling
> I saw in your loom and also tried myself - after finishing stripe payment the `plan` is `none` - shouldn't it be one of the `PlatformPlan` enum values?
> 
> <img alt="Screenshot 2024-05-08 at 19 13 01" width="748" src="https://private-user-images.githubusercontent.com/42170848/328981249-6702ba17-1dfe-4b71-9830-1e408e219f4c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUyNDk3ODIsIm5iZiI6MTcxNTI0OTQ4MiwicGF0aCI6Ii80MjE3MDg0OC8zMjg5ODEyNDktNjcwMmJhMTctMWRmZS00YjcxLTk4MzAtMWU0MDhlMjE5ZjRjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTA5VDEwMTEyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM1ZjYyNWVjOGQ3MzcwN2U3YmYwNDM2ZjI2MWZmZmFmMDY1ZGY5YTBhZTM3YmZjN2NlZGRkMjZmM2ZhYWNmOGUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.kSFJDg9UMHUWOB397Y-sfc5LgMSuk5b0T_s4usNc1i4">
> ## [LOW] Sidebar
> Personally, I would remove "Cal.com atoms" entry from sidebar and just rename "Platform docs" to "Documentation". Also, instead of having "Examples app" in the sidebar I would just put it in docs - I don't know how much value do people have of having a link to examples app because docs should mention it.
> 
> <img alt="Screenshot 2024-05-08 at 19 17 28" width="1706" src="https://private-user-images.githubusercontent.com/42170848/328981132-59ba867c-fbcc-4d2b-858d-a0601cbb4361.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUyNDk3ODIsIm5iZiI6MTcxNTI0OTQ4MiwicGF0aCI6Ii80MjE3MDg0OC8zMjg5ODExMzItNTliYTg2N2MtZmJjYy00ZDJiLTg1OGQtYTA2MDFjYmI0MzYxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTA5VDEwMTEyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZhMGEwYmIwOWY3YTI1MjYwNmVkMjc2OTQ1ZTZlNjg3ZTRkMmI0M2M3ZDY5NTAxZGIyYWI2ZDVjOWYyODlmMmQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ixOj9jJW5o9NlVRHBJEC81TOSi1FsM2rouQOa6tqCFM">
> ## [HIGH] OAuth client users not appearing?
> I created OAuth client and then created managed user for it via API- managed user email has the same OAuth client ID as in dashboard below. However, after creating the user I refresh "settings/platform" and the managed user is not appearing.
> 
> Create managed user: <img alt="Screenshot 2024-05-08 at 19 24 48" width="1285" src="https://private-user-images.githubusercontent.com/42170848/328982033-9cde4ec8-d90c-41d0-84e4-fbf353b7ae85.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUyNDk3ODIsIm5iZiI6MTcxNTI0OTQ4MiwicGF0aCI6Ii80MjE3MDg0OC8zMjg5ODIwMzMtOWNkZTRlYzgtZDkwYy00MWQwLTg0ZTQtZmJmMzUzYjdhZTg1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTA5VDEwMTEyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTkzOTBhZjQ4YjZiYmUxZWFhNmU0M2M1MWQ1OTk1YjBiZGY5ODRiYzFhOTBlNDY1MDk5ZDVjMmZjNWRlN2Y5YzgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.N2Hpmi9fv_1OWAWI7tSPeqSeaUY8ZCpBgFMoHlAkNxg">
> 
> It's not appearing: <img alt="Screenshot 2024-05-08 at 19 26 35" width="1709" src="https://private-user-images.githubusercontent.com/42170848/328982145-6a4e158b-e9d1-43ac-85da-97fe6a860b97.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUyNDk3ODIsIm5iZiI6MTcxNTI0OTQ4MiwicGF0aCI6Ii80MjE3MDg0OC8zMjg5ODIxNDUtNmE0ZTE1OGItZTlkMS00M2FjLTg1ZGEtOTdmZTZhODYwYjk3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTA5VDEwMTEyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMyNzllY2YyOTYxMDg5YmQ2OGVlYzdiYjhlZDIwMTljNTJiM2I4NzFlM2NiMWJlOWM5ZTlkZWU2ZjUwMTI3ZmImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.GLmETNsM67q2hdB94lTXV24flkKXXyW3_gKbCgaV7kA">
> 
> Does OAuth client users show up for you under the "Managed users" ?

- The plan column was not working because you need to configure the stripe webhook which can be done through the stripe CLI and once thats configured it works as expected
- Removed examples app from sidebar and renamed platform docs to documentation so we're good on that
- Managed users appears now since we have a different endpoint for it

Ryukemeister committed:
Merge branch 'main' into platform-onboarding
